app/models/Business.scala

```diff
@@ -0,0 +1,98 @@
+package models
+
+import play.api.libs.json._
+import BusinessFields._
+
+import play.api.libs.functional.syntax._
+
+case class Business(
+    id: Long,
+    businessName: String,
+    uprn: Option[Long],
+    postCode: Option[String],
+    industryCode: Option[String],
+    legalStatus: Option[String],
+    tradingStatus: Option[String],
+    turnover: Option[String],
+    employmentBands: Option[String],
+    vatRefs: Option[Seq[String]],
+    payeRefs: Option[Seq[String]],
+    companyNo: Option[String]
+) {
+  // method that used as output on UI (some fields are hidden)
+  def secured: Business = this.copy(vatRefs = None, payeRefs = None, uprn = None)
+}
+
+object Business {
+
+  def mapStrOption(opt: Option[String]): Option[String] = opt match {
+    case Some(str) if (!str.trim.isEmpty) => opt
+    case _ => None
+  }
+
+  implicit val businessReads = (
+    (JsPath \ "id").read[Long] and
+    (JsPath \ "BusinessName").read[String] and
+    (JsPath \ "UPRN").readNullable[Long] and
+    (JsPath \ "PostCode").readNullable[String] and
+    (JsPath \ "IndustryCode").readNullable[String] and
+    (JsPath \ "LegalStatus").readNullable[String] and
+    (JsPath \ "TradingStatus").readNullable[String] and
+    (JsPath \ "Turnover").readNullable[String] and
+    (JsPath \ "EmploymentBands").readNullable[String] and
+    (JsPath \ "VatRefs").readNullable[Seq[String]] and
+    (JsPath \ "PayeRefs").readNullable[Seq[String]] and
+    (JsPath \ "CompanyNo").readNullable[String]
+  )((id, businessName, uprn, postcode, industryCode, legalstatus, radingstatus, turnover, employmentbands, vatrefs, payerefs, companyno) =>
+      Business.apply(id, businessName, uprn, mapStrOption(postcode), mapStrOption(industryCode),
+        mapStrOption(legalstatus), mapStrOption(radingstatus), mapStrOption(turnover), mapStrOption(employmentbands),
+        vatrefs,
+        payerefs,
+        mapStrOption(companyno)))
+
+  implicit val biWrites = new Writes[Business] { //writes use only for hbase caching
+    override def writes(b: Business): JsValue = {
+      import b._
+      JsObject(Seq(
+        "id" -> Json.toJson(id),
+        "BusinessName" -> Json.toJson(businessName),
+        "UPRN" -> Json.toJson(uprn),
+        "PostCode" -> Json.toJson(postCode.getOrElse("")),
+        "IndustryCode" -> Json.toJson(industryCode.getOrElse("")),
+        "LegalStatus" -> Json.toJson(legalStatus.getOrElse("")),
+        "TradingStatus" -> Json.toJson(tradingStatus.getOrElse("")),
+        "Turnover" -> Json.toJson(turnover.getOrElse("")),
+        "EmploymentBands" -> Json.toJson(employmentBands.getOrElse("")),
+        vatRefs.map(v => ("VatRefs" -> Json.toJson(v.filterNot(_.trim.isEmpty)))).orNull,
+        payeRefs.map(pr => ("PayeRefs" -> Json.toJson(pr.filterNot(_.trim.isEmpty)))).orNull,
+        "CompanyNo" -> Json.toJson(companyNo.getOrElse(""))
+      ).filter(_ != null))
```

Comment:
null is a bit ugly for Scala.
If the fields are options - then by default a writer will omit a None.  Not sure how it works when explicitly creating a JsObject in this manner though ...